src/MomentoCacheClient.php

```
@@ -1443,11 +1457,69 @@ public function select(int $db): Redis|bool
+            if (isset($options['ex'])) {
+                $ttl = $options['ex'];
+            } elseif (isset($options['px'])) {
+                $ttl = $options['px'] / 1000;
+            } elseif (isset($options['exat'])) {
+                $ttl = $options['exat'] - time();
+            } elseif (isset($options['pxat'])) {
+                $ttl = floor(($options['pxat'] - microtime(true) * 1000) / 1000);
+            } elseif (isset($options['keepttl'])) {
+                throw MomentoToPhpRedisExceptionMapper::createArgumentNotSupportedException('set', 'keepttl');
+            }
+
+            // Handle NX option: Set if the key does not exist
+            if (in_array('nx', $options, true)) {
+                $result = $this->client->setIfAbsent($this->cacheName, $key, $value, $ttl);
+                if ($result->asStored()) {
+                    return "OK";
+                } else if ($result->asNotStored()) {
+                    return false;
+                } else {
+                    logger()->error('Unexpected result from setnx', ['result' => $result]);
+                    return false;
+                }
+            }
+
+            // Handle XX option: Set only if the key already exists
+            if (in_array('xx', $options, true)) {
+                $result = $this->client->setIfPresent($this->cacheName, $key, $value, $ttl);
+                if ($result->asStored()) {
+                    return "OK";
+                } else if ($result->asNotStored()) {
+                    return false;
+                } else {
+                    logger()->error('Unexpected result from setxx', ['result' => $result]);
+                    return false;
+                }
+            }
+
+            // Handle GET option: Return the previous value and set the new value
+            if (in_array('get', $options, true)) {
+                throw MomentoToPhpRedisExceptionMapper::createArgumentNotSupportedException('set', 'get');
+            }
```

We should check for this at the top to fail fast